backends/alpm/pk-alpm-packages.c

```diff
@@ -304,17 +306,13 @@ pk_backend_get_files_thread (PkBackendJob *job, GVariant* params, gpointer p)
 		if (pkg == NULL)
 			break;
 
-		root = alpm_option_get_root (priv->alpm);
-		files = g_string_new ("");
-
 		filelist = alpm_pkg_get_files (pkg);
+		files = g_new0 (gchar*, filelist->count + 1);
```

Comment:
Does this code is leaking memory at each iteration or _cleanup_strv_free_ called each time the end of loop block reached ?